### ISSUE TITLE
FR: Flag `--stdio` for `rclone serve sftp`

### ISSUE THREAD

**Issue by tomfitzhenry** [rclone/rclone#5311]
#### What is your current rclone version (output from `rclone version`)?
restic 1.53

#### What problem are you are trying to solve?

I'd like to expose a subdirectory `photos` in my home directory to a semi-trusted device, over SFTP.

To do this, I'd add the following to ~/.ssh/authorized_keys:

```
restrict,command="rclone serve sftp --stdio ./photos" ssh-rsa ...
```

Then, that key could be used by my semi-trusted device to speak sftp to just ~/photos.

The protocol would be the same stdin/stdout one as in https://man7.org/linux/man-pages/man8/sftp-server.8.html

#### How do you think rclone should be changed to solve that?
`rclone serve sftp` should have a flag `--stdio`, similar to what `rclone serve restic` has.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by tomfitzhenry** [comment]
Possibly just a matter of wiring https://github.com/rclone/rclone/blob/6ef7178ee4465f2361b6e2cf483cef5bbfd9c95c/cmd/serve/restic/stdio_conn.go#L22 (a net.Conn) to https://github.com/rclone/rclone/blob/6ef7178ee4465f2361b6e2cf483cef5bbfd9c95c/cmd/serve/sftp/server.go#L83 ?

**Comment by tomfitzhenry** [comment]
No, https://github.com/rclone/rclone/blob/6ef7178ee4465f2361b6e2cf483cef5bbfd9c95c/cmd/serve/sftp/server.go#L83 would be the wrong place to hook in, since server.acceptConnection(net.Conn) begins the SSH handshake. https://github.com/rclone/rclone/blob/6ef7178ee4465f2361b6e2cf483cef5bbfd9c95c/cmd/serve/sftp/connection.go#L261 is likely the better entrypoint.

But it's might not be as simple as just wiring it to stdin/stdout.

Ultimately `rclone serve sftp` would have to re-create the stdin/stdout protocol that OpenSSH uses when passing subsystem sftp via ForceCommand. I can't find any documentation on this.

**Comment by ncw** [comment]
> Ultimately `rclone serve sftp` would have to re-create the stdin/stdout protocol that OpenSSH uses when passing subsystem sftp via ForceCommand. I can't find any documentation on this.

I'd imagine (after doing no research) that it is just a single ssh stream (equivalent to a TCP connection) and if the ssh server needs more sftp servers then it will have to open another one.

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.56

Thank you @tomfitzhenry for implementing it :-)
